Client/Frontend/Browser/TabTrayController.swift

```diff
@@ -386,12 +399,21 @@ class TabTrayController: UIViewController, UITabBarDelegate, UITableViewDelegate
 
         let screenshotAspectRatio = tableView.frame.width / TabTrayControllerUX.CellHeight
         cell.background.image = screenshotHelper.takeScreenshot(tab, aspectRatio: screenshotAspectRatio, quality: 1)
+        cell.closeTab.addTarget(self, action: "SELdidPressClose:", forControlEvents: UIControlEvents.TouchDown)
+
         return cell
     }
 
     func tableView(tableView: UITableView, editingStyleForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCellEditingStyle {
         return .None
     }
+
+    func SELdidPressClose(sender: UIButton) {
+        let indexPath:NSIndexPath = NSIndexPath(forRow: sender.tag, inSection: 0)
```

Comment:
It doesn't look like `tag` gets set anywhere, so this won't work. It will always be 0.